src/features/bookings/helpers/expirationDateUtils.test.ts

```diff
@@ -0,0 +1,80 @@
+import { bookingsSnap } from 'features/bookings/fixtures/bookingsSnap'
+import {
+  displayExpirationMessage,
+  getDigitalBookingWithoutExpirationDate,
+  isBookingInList,
+} from 'features/bookings/helpers/expirationDateUtils'
+import { Booking } from 'features/bookings/types'
+
+describe('expirationDateUtils', () => {
+  const initialBookings: Booking[] = bookingsSnap.ongoing_bookings
+  describe('getDigitalBookingWithoutExpirationDate', () => {
+    it('should get an array with a booking if is digital and without expiration date', () => {
+      // isDigital === true && !booking.expirationDate
+      const validBooking = initialBookings[0]
+      const arrayOfBooking = [validBooking]
+
+      expect(getDigitalBookingWithoutExpirationDate(initialBookings)).toStrictEqual(arrayOfBooking)
+    })
+
+    it('should return an empty array ', () => {
+      // isDigital === false && !booking.expirationDate
+      const invalidbooking = {
+        ...initialBookings[1],
+        stock: {
+          ...initialBookings[1].stock,
+          offer: {
+            ...initialBookings[1].stock.offer,
+            isDigital: false,
+          },
+        },
+      }
+      const arrayOfBooking = [invalidbooking]
+
+      expect(getDigitalBookingWithoutExpirationDate(arrayOfBooking)).toEqual([])
+    })
+  })
+
+  describe('isBookingInList', () => {
+    it('should check if a booking does exist in the list of DigitalBookingWithoutExpirationDate array and return true if it exist', () => {
+      const [firstBooking] = initialBookings
+      const getDigitalBookingWithoutExpirationDate = initialBookings
+
+      expect(isBookingInList(firstBooking, getDigitalBookingWithoutExpirationDate)).toBeTruthy()
+    })
+
+    it('should check if a booking does exist in the list of DigitalBookingWithoutExpirationDate array and return false if it does not exist ', () => {
+      const [firstBooking] = initialBookings
+      const newFirstBooking = { ...firstBooking, id: 999 }
+      const getDigitalBookingWithoutExpirationDate = initialBookings
+
+      expect(isBookingInList(newFirstBooking, getDigitalBookingWithoutExpirationDate)).toBeFalsy()
+    })
+  })
+
+  describe('displayExpirationMessage', () => {
```

Comment:
La question précédente peut aussi être vérifié ici dans un premier lieu.

Reply:
Dans ce cas la je l'ai déjà vérifié dans le dernier test  😃  :
` it('should display nothing when daysLeft < 0', () => {
      const daysLeft = -1

      expect(displayExpirationMessage(daysLeft)).toBe('')
    })`